### PR TITLE
Add manual deploy workflow for zcashd images

### LINKED ISSUE #1222: Deploy more Zebra or zcashd instances on testnet
## Motivation

The Zebra sync tests currently run on mainnet and testnet. Sometimes the testnet nodes are slow or unreliable, so the sync tests are unreliable.

## Solution

Deploy about 10 Zebra or `zcashd` instances on testnet, to increase the average speed and reduce the average latency of testnet.

## Open Questions

Right now, we just need a few more Zebra or `zcashd` instances to stabilise the test network. We can deal with distribution, mining, and network speed later.

Should the `zcashd` instances be configured to mine blocks?
  - Maybe a few should mine on 1 core each

Where should they be located? How fast should they be?
  - Reasonably fast
  - Distributed across multiple DCs
  - In reasonably central internet locations, so Zebra developers have low latency for quick sync tests
  - Near our CI and CD runners?

## Related Issues

Testnet sync fixes: Issue #1207 PR #1223
Tracking issue #745

### PR DESCRIPTION
**Pull request by yaahc**
## Motivation

The Zebra sync tests currently run on mainnet and testnet. Sometimes the testnet nodes are slow or unreliable, so the sync tests are unreliable.

## Solution

Deploy about 10 Zebra or zcashd instances on testnet, to increase the average speed and reduce the average latency of testnet. This PR adds a github action workflow for deploying a group of zcashd nodes

The code in this pull request has:
  - [x] Documentation Comments

## Related Issues

https://github.com/ZcashFoundation/zebra/issues/1222

### CODE REVIEW

**Comment by yaahc** [comment]
deployment successfully tested from my fork: https://github.com/yaahc/zebra/runs/2035890120?check_suite_focus=true

File: .github/workflows/zcashd-cd.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    inputs:
+      network:
+        default: 'testnet'
+      size:
+        default: 10
+
+env:
+  PROJECT_ID: zealous-zebra
+  REGION: us-east1
+  ZONE: us-east1-b
```
Comment (dconnolly) [inline .github/workflows/zcashd-cd.yml]: us-east1 sometimes gets full with our stuff / approaches quota with our existing zebrad deployments, so I'd suggest a different region:

```suggestion
  REGION: us-central1
  ZONE: us-central1-c
```

File: .github/workflows/zcashd-cd.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      # Create instance template from container image
+      - name: Create instance template
+        run: |
+          gcloud compute instance-templates create-with-container "zcashd-$BRANCH_NAME-$SHORT_SHA" \
+          --container-image "electriccoinco/zcashd" \
+          --container-env ZCASHD_NETWORK="${{ github.event.inputs.network }}" \
+          --machine-type n2d-standard-4 \
+          --service-account cos-vm@zealous-zebra.iam.gserviceaccount.com \
+          --scopes cloud-platform \
+          --tags zcashd \
```
Comment (dconnolly) [inline .github/workflows/zcashd-cd.yml:49]: I just updated the firewall rule to allow instances with this tag to also allow incoming traffic on the 8233, 18233 ports, same as the `zebrad` tag

File: .github/workflows/zcashd-cd.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        with:
+          version: '295.0.0'
+          project_id: ${{ env.PROJECT_ID }}
+          service_account_key: ${{ secrets.GCLOUD_AUTH }}
+
+      # Create instance template from container image
+      - name: Create instance template
+        run: |
+          gcloud compute instance-templates create-with-container "zcashd-$BRANCH_NAME-$SHORT_SHA" \
+          --container-image "electriccoinco/zcashd" \
```
Comment (dconnolly) [inline .github/workflows/zcashd-cd.yml:44]:
```suggestion
          --boot-disk-size 100GB \
          --container-image "electriccoinco/zcashd" \
```

File: .github/workflows/zcashd-cd.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set project and image names
+        run: |
+          BRANCH_NAME=$(expr $GITHUB_REF : '.*/\(.*\)') && \
+          BRANCH_NAME=${BRANCH_NAME,,} && \
+          REPOSITORY=${GITHUB_REPOSITORY,,} && \
```
Comment (dconnolly) [inline .github/workflows/zcashd-cd.yml]:
```suggestion
```

File: .github/workflows/zcashd-cd.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          echo "BRANCH_NAME=$BRANCH_NAME" >> $GITHUB_ENV && \
+          echo "SHORT_SHA=$(git rev-parse --short=7 $GITHUB_SHA)" >> $GITHUB_ENV && \
+          echo "REPOSITORY=$REPOSITORY" >> $GITHUB_ENV
```
Comment (dconnolly) [inline .github/workflows/zcashd-cd.yml]:
```suggestion
```

**Comment by yaahc** [comment]
tested again successfully on fork: https://github.com/yaahc/zebra/runs/2042664696?check_suite_focus=true

**Review by dconnolly — APPROVED** [review]

**Pull request merged**